### DISCUSSION TITLE
API - Unable to login through Web Application Firewall - "The Semaphore timeout perdiod has expired"

### DISCUSSION THREAD

**Discussion by Marc-Harony** [rustdesk/rustdesk-server-pro#381] [General]
Hi! 👋🏻 

I'm running the WEB Console behind a WAF to get SSL encryption and for a better security.

I don't want to expose the web console port to the WAN.

I've set the URL "https://rustdesk.domain.com" in the API settings, but when I'm trying to log into my account, I have the following error:
![image](https://github.com/user-attachments/assets/9aae7d95-2552-4434-8704-99d63a689d50)

More info about the rustdesk server:

I'm running the dockerized version of hbbs and hbbr.
There's a NAT rule for 21115, 21116 and 21117.
There's a WAF rule to access the webconsole.
Login into the client app is working when the port 21114 is added to the NAT rule but that's not a thing I want.

Thanks for the help you can bring 🙏🏻

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/wiki/FAQ#failed-to-log-in-with-custom-client-build-or-public-release-with-wrong-api-settings

**Reply by rustdesk** [reply]
![image](https://github.com/user-attachments/assets/5f9270e8-fd3e-4a38-8ae4-faabd6dcabdc)

**Reply by Marc-Harony** [reply]
Well, hehe, thanks!

I feel dumb now 😅

**Comment by Marc-Harony** [comment]
Closing as the problem is solved.

Again, thanks for your help.
